### PR TITLE
LPD-22945: Fix SOLR adapter Boolean query translation

### PR DESCRIPTION
**Pull request by peerkar**
- https://liferay.atlassian.net/browse/LPD-22379
  - https://liferay.atlassian.net/browse/LPD-22945

### CODE REVIEW

**Comment by peerkar** [comment]
ci:test:sf

**Comment by peerkar** [comment]
@gustavolimav actually this is in @BryanEngler 's review

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/query/BooleanQueryTranslatorImpl.java [hunk 1]
```diff
@@ -30,32 +34,43 @@ public org.apache.lucene.search.Query translate(
...
 
 			if (query != null) {
-				builder.add(
+				booleanQueryBuilder.add(
 					query, translate(booleanClause.getBooleanClauseOccur()));
 			}
 		}
 
-		org.apache.lucene.search.Query query = builder.build();
+		BooleanFilter booleanFilter = booleanQuery.getPreBooleanFilter();
```
Comment (BryanEngler) [inline modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/query/BooleanQueryTranslatorImpl.java:52]: I know you didn't create this API, but im just curious if you have any idea why a Boolean Query has a "Pre Boolean Filter"?

Maybe `BooleanQuery extends Query` was a mistake...

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/query/translator/BooleanQueryTranslatorImpl.java [hunk 2]
```diff
@@ -24,6 +24,10 @@ public org.apache.lucene.search.Query translate(
 		org.apache.lucene.search.BooleanQuery.Builder builder =
 			new org.apache.lucene.search.BooleanQuery.Builder();
 
+		processQueryClause(
```
Comment (BryanEngler) [inline modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/query/translator/BooleanQueryTranslatorImpl.java:27]: does this mean we should also process `booleanQuery.getShouldQueryClauses()`?

File: modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/search/engine/adapter/search/BaseSolrQueryAssemblerImpl.java [hunk 3]
```diff
@@ -91,11 +90,11 @@ protected void addFilterQuery(
 			return;
 		}
 
-		String filterString = translate(booleanClause);
+		org.apache.lucene.search.Query filterQuery = translate(booleanClause);
 
 		filterQueries.add(
 			StringBundler.concat(
-				"{!tag=", tag, StringPool.CLOSE_CURLY_BRACE, filterString));
+				"{!tag=", tag, StringPool.CLOSE_CURLY_BRACE, filterQuery));
```
Comment (BryanEngler) [inline modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/search/engine/adapter/search/BaseSolrQueryAssemblerImpl.java:97]: small sf, just pass in translate(booleanClause)?

**Comment by BryanEngler** [comment]
overall looks good, did you run the solr suite against it?

i think some tests were added for https://liferay.atlassian.net/browse/LPS-154206, should double check those still pass since you deleted the coded that was added with those tests.

**Comment by peerkar** [comment]
ci:test:search-solr8

**Comment by peerkar** [comment]
ci:test:search-solr

**Comment by peerkar** [comment]
ci:test:sf

**Comment by peerkar** [comment]
ci:test:stable

**Comment by peerkar** [comment]
ci:reevaluate:1267401_13155

**Comment by peerkar** [comment]
ci:stop

**Comment by peerkar** [comment]
CI failures unrelated. Sending directly [here](https://github.com/brianchandotcom/liferay-portal/pull/149491)

**Pull request closed without merging**